provisioner/file/provisioner.go

```diff
@@ -154,7 +159,11 @@ func (p *Provisioner) ProvisionUpload(ui packer.Ui, comm packer.Communicator) er
 			return err
 		}
 
-		err = comm.Upload(p.config.Destination, f, &fi)
+		if strings.HasSuffix(dst, "/") {
+			dst = filepath.Join(dst, filepath.Base(src))
```

Comment:
hmm, I can't reproduce that behavior. https://gist.github.com/mwhooker/d03c83365558f1a459b6fc2fc495e122